app/scripts/modules/core/src/pipeline/config/stages/managed/ImportDeliveryConfigExecutionDetails.tsx

```diff
@@ -2,10 +2,69 @@ import React from 'react';
 
 import { ExecutionDetailsSection, IExecutionDetailsSectionProps, StageFailureMessage } from 'core/pipeline';
 import { IGitTrigger } from 'core/domain';
+import { CollapsibleSection, Markdown } from 'core/presentation';
 import { SETTINGS } from 'core/config';
 
 const NOT_FOUND = 'Not found';
 
+// Turn off linter for the block below because it wants CamelCase property names.
+/* eslint-disable @typescript-eslint/camelcase */
+const ERROR_MESSAGE_MAP: any = {
+  missing_property: 'The following property is missing:',
+  invalid_type: 'The type of the following property is invalid:',
+  invalid_format: 'The format of the following property is invalid:',
+  invalid_value: 'The value of the following property is invalid:',
+};
+/* eslint-enable */
+
+interface IDeliveryConfigImportErrorDetails {
+  error: string;
+  message: string;
+  pathExpression: string;
+}
+
+interface IDeliveryConfigImportError {
+  message: string;
+  details?: IDeliveryConfigImportErrorDetails;
+}
+
+const CustomErrorMessage = (message: string, debugDetails?: string) => {
+  return (
+    <div>
+      <div className="alert alert-danger">
+        <b>There was an error parsing your delivery config file.</b>
+        <br />
+        <Markdown message={message} style={{ wordBreak: 'break-all' }} />
+        <br />
+        {debugDetails && (
+          <CollapsibleSection heading={({ chevron }) => <span>{chevron} Debug Details</span>}>
+            <pre style={{ whiteSpace: 'pre-wrap' }}>{debugDetails}</pre>
+          </CollapsibleSection>
+        )}
+      </div>
+    </div>
+  );
+};
+
+function buildCustomErrorMessage(error: IDeliveryConfigImportError) {
+  if (!error) {
+    return null;
+  }
+
+  if (!error.details) {
+    return CustomErrorMessage(error.message);
+  }
+
+  // Replace dots with slashes for paths because it feels more familiar. Also ditch the very first slash/dot as it just adds noise.
+  const pathExpression = error.details.pathExpression.substring(1).replace(/\./g, '/');
+
+  const errorMessage = ERROR_MESSAGE_MAP[error.details.error]
+    ? ERROR_MESSAGE_MAP[error.details.error] + `<br/>\`${pathExpression}\``
+    : 'Unknown error';
+
+  return CustomErrorMessage(errorMessage, error.details.message);
```

Comment:
In addition to turning `CustomErrorMessage` into a React component, I recommend decoupling this `buildCustomErrorMessage` function from `CustomErrorMessage`. Instead, consider returning the error message/details as data and then passing that data to the component. For example instead of the code currently on line 99 you might write:

```tsx
// maybe buildErrorMessage returns an object like { summary: string, debugDetails: string }?
const errorMessage = buildErrorMessage(stage.context.error as IDeliveryConfigImportError);

{errorMessage ? <CustomErrorMessage message={errorMessage.summary} debugDetails={errorMessage.debugDetails} /> : <StageFailureMessage stage={stage} />}
```

Reply:
Done!